cmd/gke-gcloud-auth-plugin/gcloud_token_provider.go

```diff
@@ -25,11 +25,20 @@ type gcloudConfiguration struct {
 type gcloudTokenProvider struct {
 	readGcloudConfigRaw func() ([]byte, error)
 	readFile            func(filename string) ([]byte, error)
+	account             string
+	project             string
 }
 
 // readGcloudConfig returns an object which represents gcloud config output
 func (p *gcloudTokenProvider) readGcloudConfig() (*gcloudConfiguration, error) {
-	gcloudConfigBytes, err := p.readGcloudConfigRaw()
+	extraArgs := []string{}
+	if account != "" {
+		extraArgs = append(extraArgs, fmt.Sprintf("--account=%s", account))
+	}
+	if project != "" {
```

Comment:
Can you please add impersonateServiceAccount as well?

Reply:
Since that one was not originally passed through to the non edge cloud commands I wanted to keep the changes to within the scope we needed, If you are pretty sure passing it along won't cause any potential issues or what not then I have no issues with it. I don't know the reason it was only passed to edge cloud so I didn't want to touch it since it is outside the scope for this change.

Reply:
Yes, please pass it! I am sure regular gcloud folks will want to use it as well!

Reply:
Please add location as well, since the flag already exists!!